cornac/models/gp_top/recom_gp_top.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 The Cornac Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+
+import numpy as np
+from collections import Counter
+
+from ..recommender import NextBasketRecommender
+
+
+class GPTop(NextBasketRecommender):
+    """Global Personalized Top Frequent Items.
+
+    Parameters
+    ----------
+    name: string, default: 'GPTop'
+        The name of the recommender model.
+
+    use_global_popularity: boolean, optional, default: True
+        When False, no item frequency from all users' baskets are being used.
+
+    use_personalized_popularity: boolean, optional, default: True
+        When False, no item frequency from history baskets are being used.
+
+    References
+    ----------
+    Ming Li, Sami Jullien, Mozhdeh Ariannezhad, and Maarten de Rijke. 2023.
+    A Next Basket Recommendation Reality Check.
+    ACM Trans. Inf. Syst. 41, 4, Article 116 (October 2023), 29 pages. https://doi.org/10.1145/3587153
+
+    """
+
+    def __init__(
+        self, name="GPTop", use_global_popularity=True, use_personalized_popularity=True
+    ):
+        super().__init__(name=name, trainable=False)
+        self.use_global_popularity = use_global_popularity
+        self.use_personalized_popularity = use_personalized_popularity
+        self.item_freq = Counter()
+
+    def fit(self, train_set, val_set=None):
+        super().fit(train_set=train_set, val_set=val_set)
+        if self.use_global_popularity:
+            self.item_freq = Counter(self.train_set.uir_tuple[1])
```

Comment:
it seems that we only need `max_item_freq` and don't have to store the frequency for all items. We can just do it with 
```python
np.bincount(self.train_set.uir_tuple[1]).max()
```